File: 1869-longer-contiguous-segments-of-ones-than-zeros/1869-longer-contiguous-segments-of-ones-than-zeros.cs
public class Solution {
    public bool CheckZeroOnes(string s) {
        
        int zCount = 0 , oCount = 0;
        
        int current = 0;
        
        if (s=="1")
            return true;
        
        for (int i = 0; i < s.Length; i++){
            if (i>0 && s[i] == s[i-1]){
                current += 1;
                if (s[i] == '1'){
                    
                    oCount = Math.Max(oCount, current);
                }
                else if(s[i] == '0'){
                    zCount = Math.Max(zCount, current);
                }
            }
            else{
                current = 0;
            }
        }
        
        return zCount < oCount;
        
    }
}

File: 1869-longer-contiguous-segments-of-ones-than-zeros/1869-longer-contiguous-segments-of-ones-than-zeros.py
class Solution:
    def checkZeroOnes(self, s: str) -> bool:
        long_0 = 0
        long_1 = 0
        
        count = 0
        
        prev = '-'
        
        if s == "1": return True
        
        for ch in s:
            if ch == prev:
                count+= 1
                if ch == "1":
                    long_1 = max(long_1, count)
                elif ch == "0":
                    long_0 = max(long_0, count)
            else:
                count = 0
            
            prev = ch
        
        return long_1 > long_0